core/src/main/java/com/kumuluz/ee/EeApplication.java

```diff
@@ -106,13 +110,27 @@ private void initialize() {
         eeConfig.getEeComponents().addAll(eeComponents);
 
         // Loading the extensions and extracting its metadata (
-        // required - @EeExtensionDef anotation)
-        // Check for EE dependencies on extensions
+        List<Extension> extensions = ExtensionLoader.loadExtensions();
 
         // Initiate the config extensions (filter(c.type -> c == CONFIG))
-        // Initialize ConfigurationDispatcher
-        // getPropery()
-        // Insert into ConfigurationUtil
+        log.info("Initializing config extensions");
+
+        List<Extension> configExtensions = extensions.stream().filter(extension -> extension.getClass()
+                .getDeclaredAnnotation(EeExtensionDef.class).type().equals(EeExtensionType.CONFIG)).collect
+                (Collectors.toList());
+
+        for (Extension configExtension : configExtensions) {
```

Comment:
Additionally check if the extension contains a list of required Ee dependencies and validate if they are fulfilled.